exchange/index.go

```diff
@@ -453,6 +490,165 @@ func (idx *Index) evict(size uint64) uint64 {
 	return evicted
 }
 
+func (idx *Index) tagForEviction(ref *DataRef) error {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	err := idx.browseDAG(ref.PayloadCID, func(blk blocks.Block) error {
+		idx.evictionSet.Add(blk.Cid())
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (idx *Index) GC() {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	// exit if there is nothing to evict
+	if idx.evictionSet.Len() == 0 {
+		return
+	}
+
+	unlock := idx.bstore.GCLock()
+	defer unlock.Unlock()
+
+	err := idx.evictionSet.ForEach(func(c cid.Cid) error {
+		return idx.bstore.DeleteBlock(c)
+	})
+	if err != nil {
+		log.Error().Err(err).Msg("failed to run garbage collector")
+	}
+
+	idx.evictionSet = cid.NewSet()
+	idx.lastEvictionTime = time.Now()
+}
+
+func (idx *Index) gcLoop() {
```

Comment:
Can you remind me why we need that loop again? It just seems like the index should expose GC methods but not execute them.

Reply:
Because we run the GC only when we commit, if we don't commit anything but we tag nodes for eviction, they will never be removed, so there is a loop that runs every 5 min to evict these tagged nodes

Reply:
We only tag when we commit as well though so I think we might be fine without it

Reply:
Hum we tag when we drop or Set a ref

Reply:
The node currently does not expose a method for users to delete content, it is only evicted when new content is added over the storage capacity. That new content is added once you call commit() then set the ref of the committed transaction. Which is when we would trigger GC. 
I don't see a situation where a periodic GC is currently needed but my head is in other stuff so if I'm missing something can you explain under which condition storage might significantly overflow?

Reply:
Yeah you're right we don't need that, I removed it